internal/dataplane/parser/translate_httproute.go

```diff
@@ -127,9 +127,10 @@ func generateKongRoutesFromHTTPRouteRule(httproute *gatewayv1alpha2.HTTPRoute, r
 			// determine the name of the route, identify it as a route that belongs
 			// to a Kubernetes HTTPRoute object.
 			routeName := kong.String(fmt.Sprintf(
-				"httproute.%s.%s.%d",
+				"httproute.%s.%s.%d.%d",
```

Comment:
Minor consistency bikeshedding: for Ingress we don't separate the rule and match numbers with `.`: https://github.com/Kong/kubernetes-ingress-controller/blob/1c05a64be71bce65bd21be3a2514e90586a5623e/internal/dataplane/parser/translate_ingress.go#L60

```suggestion
				"httproute.%s.%s.%d%d",
```

Reply:
While I understand that this diverges from `Ingress` the older behavior I expect is actually kinda broken at scales beyond just a few rules. If rules stay under 10 count then things more or less make sense, but past that what does 1111 mean here? It just seems to get kinda fuzzy as you go up, where this way is delineated and so perfectly clear. I feel strongly that given the opportunity we have here to clearly delineate we should take it, and given your PR approval it seems you feel OK with either choice. However I understand the implications of what you're saying, and not continuing to do things that way does cause some divergence. As such, I created a new label called `area/maintenance (breaking)` which is described as `Maintenance requiring backwards-incompat changes (used to track the need for a new major release)`. This is effectively a bucket where we can put changes that should (or that we want to see) happen but can't happen in the current major release cycle, and thus can help us to track a possible future need for a KIC V3. I've added #2347 to it which captures this problem and suggests that we should in the next major release be bringing the rest of the APIs into conformance with how we're doing `HTTPRoute` in this PR.